server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -53,13 +54,17 @@
 
     private final ContainerName containerName;
     private final org.jclouds.blobstore.BlobStore blobStore;
+    private final Function<Blob, String> putBlob;
     private final PayloadCodec payloadCodec;
 
     ObjectStorageBlobsDAO(ContainerName containerName, BlobId.Factory blobIdFactory,
-                          org.jclouds.blobstore.BlobStore blobStore, PayloadCodec payloadCodec) {
+                          org.jclouds.blobstore.BlobStore blobStore,
+                          Function<Blob, String> putBlob,
```

Comment:
Use `@FunctionalInterface`

Reply:
Can the return type of the putBlob function be strong typed ?

Reply:
I think it's better to return a BlobId instead of a String